src/java.base/share/classes/java/lang/Class.java

```diff
@@ -236,11 +236,12 @@ private static void runtimeSetup() {
      * This constructor is not used and prevents the default constructor being
      * generated.
      */
-    private Class(ClassLoader loader, Class<?> arrayComponentType) {
+    private Class(ClassLoader loader, Class<?> arrayComponentType, ProtectionDomain pd) {
```

Comment:
If this constructor is not used then why do we need to add the PD argument, rather than just set it to null? For that matter why do we even need the field if nothing is ever setting it? I'm missing something here.

Reply:
@DanHeidinga suggested this for my other PR as a convention that's used for the j.l.Class constructor.

Reply:
I am still missing what can actually set a PD here, sorry. ??

Reply:
Because the field is final, it has to be initialized in the constructor in Java code.  My initial patch for modifiers chose to initialize to zero but that's not quite correct.  The constructor cannot be called nor can it be made accessible with setAccessible().   So the constructor for java.lang.Class is essentially the Hotspot code JavaClasses::create_mirror().  This is where the PD is assigned.